src/main/java/uk/nhs/ctp/auditFinder/AuditFinder.java

```diff
@@ -34,80 +38,72 @@
   private final TokenAuthenticationService authenticationService;
   private final ObjectMapper mapper;
 
-  @Override
-  public Optional<AuditSession> findByAuditId(String auditId) {
+  public List<AuditSession> findAllEncountersByOperationTypeAndInteractionId(
+      OperationType operationType, String interactionId) {
     var supplierId = authenticationService.requireSupplierId();
 
     var query = QueryBuilders.boolQuery()
         .must(QueryBuilders.termQuery(SUPPLIER_ID_FIELD, supplierId))
-        .must(QueryBuilders.termQuery(REQUEST_ID_FIELD, auditId));
-
-    var source = buildSingularSource(query);
-
-    return search(supplierId, source).collect(toOptional());
-  }
-
-  @Override
-  public List<AuditSession> findAllEncountersByCaseId(String caseId) {
-    var supplierId = authenticationService.requireSupplierId();
-
-    var query = QueryBuilders.boolQuery()
-        .must(QueryBuilders.termQuery(SUPPLIER_ID_FIELD, supplierId))
-        .must(QueryBuilders.termQuery(CASE_ID_FIELD, caseId));
+        .must(QueryBuilders.termQuery(OPERATION_FIELD, operationType.getName()))
+        .must(QueryBuilders.termQuery(INTERACTION_ID_FIELD, interactionId));
 
     var source = buildSearchSource(query);
 
     return search(supplierId, source).collect(toUnmodifiableList());
   }
 
-  @Override
   public List<AuditSession> findAllEmsEncountersByCaseId(String caseId) {
     var supplierId = authenticationService.requireSupplierId();
 
     var query = QueryBuilders.boolQuery()
         .must(QueryBuilders.termQuery(OWNER_FIELD, EMS_NAME))
         .must(QueryBuilders.termQuery(SUPPLIER_ID_FIELD, supplierId))
-        .must(QueryBuilders.termQuery(CASE_ID_FIELD, caseId));
+        .must(QueryBuilders.termQuery(INTERACTION_ID_FIELD, caseId));
 
     var source = buildSearchSource(query);
 
     return search(supplierId, source).collect(toUnmodifiableList());
   }
 
-  @Override
-  public List<AuditSession> findAllEncounters() {
+  public List<AuditSession> findInteractions() {
     var supplierId = authenticationService.requireSupplierId();
 
     var query = QueryBuilders.boolQuery()
         .must(QueryBuilders.termQuery(SUPPLIER_ID_FIELD, supplierId))
-        .must(QueryBuilders.existsQuery(CASE_ID_FIELD));
+        .must(QueryBuilders.existsQuery(INTERACTION_ID_FIELD))
+        .must(QueryBuilders.existsQuery(OPERATION_FIELD));
 
     var source = buildSearchSource(query);
 
     return search(supplierId, source).collect(toUnmodifiableList());
-  }
 
-  @Override
-  public List<AuditSession> findAllServiceSearches() {
-    var supplierId = authenticationService.requireSupplierId();
-
-    var query = QueryBuilders.boolQuery()
-        .must(QueryBuilders.termQuery(SUPPLIER_ID_FIELD, supplierId))
-        .must(QueryBuilders.termQuery(OPERATION_FIELD, OperationType.SERVICE_SEARCH.getName()));
+  }
 
-    var source = buildSearchSource(query);
+  public List<AuditInteraction> groupInteractions(List<AuditSession> auditSessions) {
```

Comment:
I would split this into it's own 'transform and group' component just for separation of concerns

Reply:
Yep, moved to `AuditTransformer.java`